### DISCUSSION TITLE
Updating a view model list with ItemsSource binding

### DISCUSSION THREAD

**Discussion by tomeverin** [xamarin/Xamarin.Forms#12120] [General]
Hi All,

Suppose I have a `List<T> DisplayItems` in my view model, and I have a collection view / bindable layout on the XAML page which has `ItemsSource="{Binding DisplayItems}"`

Suppose I want to reassign with new data, e.g. `DisplayItems = newDataListObject` in the logic of my view model.

Do I need to do that assignment on the MainThread? e.g. 
```
MainThread.BeginInvokeOnMainThread(() => {
    DisplayItems = newDataListObject;
});
````

Or will the PropertyChanged notification take care of ensuring UI is updated on the main thread?

**Comment by themronion** [comment]
I think there is no need to wrap that in BeginInvoke
